src/components/Menu.jsx

```diff
@@ -1,29 +1,70 @@
 import React, { useState, useEffect } from "react";
-import { Segment, List } from "semantic-ui-react";
+import { Segment, Grid, Menu, Container } from "semantic-ui-react";
 import MenuItem from "./MenuItem";
 import axios from "axios";
 
-const Menu = () => {
+const MenuPage = () => {
   const [menuItems, setMenuItems] = useState([]);
+  const [activeCategory, setActiveCategory] = useState("starters");
+
   useEffect(() => {
     axios.get("https://slowfood.heroku.com/api/products").then((response) => {
       setMenuItems(response.data.products);
     });
-  }, []);
+  }, [activeCategory]);
```

Comment:
Actually, I'm not sure about this at all. By removing the first `useEffect` are you simply NOT loading anything on the first render? Perhaps we should do this the other way around? Fetch all `nemuItems` on render, and then skip calling the API but rater just filter through the ones we have?

Reply:
Whenever the page loads the app makes an outgoing network call to the url, leading us to believe it will render on the first load. It also makes an outgoing network call  upon every click on the category tabs.  
![Skärmavbild 2021-10-01 kl  09 30 09](https://user-images.githubusercontent.com/29585462/135582337-fdd6ca1e-b9be-49cd-bcfd-40b7496b56ac.png)

Reply:
Yep, and you don't need to repeat request on every tab click. 1) On initial render you fetch the data 2) Save in the state 3) filter what you display on tab clicks (don't fetch again, it's unnecessary traffic and UI delay)

Reply:
We took away the callback for activeCategory and now it fills the criteria. On initial render it fetches the data and stores it in menuItems. It does not fetch more data when clicking on categories but it still filters the stored data
![Skärmavbild 2021-10-01 kl  10 16 58](https://user-images.githubusercontent.com/29585462/135588439-7917f32b-06de-47fd-b77e-e65315976283.png)
.

Reply:
correct